src/runtime_src/core/common/query_requests.h

```diff
@@ -1073,6 +1075,58 @@ struct error : request
   }
 };
 
+// Retrieve asynchronous xocl errors from xocl driver
+struct xocl_errors : request
+{
+  using result_type = std::vector<char>;
+  static const key_type key = key_type::xocl_errors;
+
+  virtual boost::any
+  get(const device*) const = 0;
+
+  // Parse sysfs line and from class get error code and timestamp
+  static std::pair<uint64_t, uint64_t>
+  to_value(const std::vector<char>& buf, xrtErrorClass ecl)
+  {
+    uint64_t errcode = 0;
+    uint64_t ts = 0;
+    if (buf.empty())
+      return std::make_pair(errcode, ts);
+    auto errors_buf = reinterpret_cast<const xcl_errors *>(buf.data());
+    if (errors_buf->num_err <= 0)
```

Comment:
Why would num_err ever be less than 0?  Should num_err have been an unsigned int?

Reply:
It is defined as integer. I think better to check for <= 0 as sysfs may get garbage sometimes :)